### PR TITLE
Update allowed dom props for components

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://jira.corp.adobe.com/browse/RSP-1783

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/provider/src/Provider.tsx [hunk 1]
```diff
@@ -61,6 +61,11 @@ function Provider(props: ProviderProps, ref: DOMRef<HTMLDivElement>) {
     ...otherProps
   } = props;
 
+  // should this be in an aria hook?
```
Comment (snowystinger) [inline packages/@react-spectrum/provider/src/Provider.tsx:64]: thoughts?

File: packages/@react-types/dialog/src/index.d.ts [hunk 2]
```diff
@@ -46,7 +46,8 @@ export interface SpectrumDialogTriggerProps extends DialogTriggerProps, Position
   isDismissable?: boolean
 }
 
-export interface SpectrumDialogProps extends DOMProps, StyleProps {
+// TODO: why would we exclude aria-hidden? nested dialogs?
```
Comment (snowystinger) [inline packages/@react-types/dialog/src/index.d.ts:49]: @jnurthen ?

File: packages/@react-types/shared/src/dom.d.ts [hunk 3]
```diff
@@ -52,8 +52,16 @@ export interface DOMProps {
...
    * Indicates that the element is perceivable but disabled, so it is not editable or otherwise operable.
+   * false is not allowed, use undefined.
+   * TODO: early comment has it as boolean | 'false' | 'true', follow up says `If possible do not allow aria-hidden=false as an allowed value`
    */
-  'aria-hidden'?: boolean | 'false' | 'true'
+  'aria-hidden'?: true | 'true'
+
+  /**
+   Indicates the element that represents the current item within a container or set of related elements.
+   TODO: is this actually supposed to be in here? one comment had it, the more recent one did not
```
Comment (snowystinger) [inline packages/@react-types/shared/src/dom.d.ts:62]: @jnurthen @majornista

File: packages/@react-types/shared/src/dom.d.ts [hunk 4]
```diff
@@ -52,8 +52,16 @@ export interface DOMProps {
 
   /**
    * Indicates that the element is perceivable but disabled, so it is not editable or otherwise operable.
+   * false is not allowed, use undefined.
+   * TODO: early comment has it as boolean | 'false' | 'true', follow up says `If possible do not allow aria-hidden=false as an allowed value`
```
Comment (snowystinger) [inline packages/@react-types/shared/src/dom.d.ts:56]: conflicts with HMTMLAttributes, thoughts?

File: packages/@react-types/shared/src/dom.d.ts (hunk 3 above)
Comment (jnurthen) [reply packages/@react-types/shared/src/dom.d.ts:62]: I think this is the best way to implement. Make a global and then disallow on things where it makes no sense

File: packages/@react-types/shared/src/dom.d.ts [hunk 5]
```diff
@@ -52,8 +52,16 @@ export interface DOMProps {
 
   /**
    * Indicates that the element is perceivable but disabled, so it is not editable or otherwise operable.
```
Comment (jnurthen) [inline packages/@react-types/shared/src/dom.d.ts:54]:
```suggestion
   * Indicates that the element is hidden from the Accessibility API. 
```

**Comment by snowystinger** [comment]
closing as this took care of this. I'll do an audit to double check it though, but this is definitely not what it would look like anymore

**Pull request closed without merging**